conftest.py

```diff
@@ -31,8 +30,7 @@ def pytest_collection_modifyitems(config, items):
 @pytest.fixture(scope="session")
 def software():
     return os.environ.get(
-        "COILED_SOFTWARE_NAME",
-        f"dask-engineering/coiled_dist-py{sys.version_info[0]}{sys.version_info[1]}",
+        "COILED_SOFTWARE_NAME", os.environ.get("DEFAULT_SOFTWARE_NAME")
```

Comment:
```suggestion
        "COILED_SOFTWARE_NAME"
```

Reply:
This works fine in CI, but means that folks will need to set an environment variable locally to run tests, which is easy to forget. I'd prefer for this to "just work" (or fail with an informative error message) these environment variables aren't set. 

For now, let's just strictly swap out the `dask-engineering/coiled_dist-py*` environments here with the newer `dask-engineering/coiled-runtime*` environments. This is already an improvement as we'll be using software environments which are automatically build for us (instead of adhoc like before). We should still think more about how we want to fix this longer-term, but let's do that separate from this PR